pkg/controllers/jobset_controller.go

```diff
@@ -431,10 +485,19 @@ func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet,
 	return nil
 }
 
-// TODO: update condition in place if it exists.
 func (r *JobSetReconciler) updateStatusWithCondition(ctx context.Context, js *jobset.JobSet, eventType string, condition metav1.Condition) error {
 	condition.LastTransitionTime = metav1.Now()
-	js.Status.Conditions = append(js.Status.Conditions, condition)
+	// If condition exists then let's overwrite.
+	conditionExists := false
+	for i, val := range js.Status.Conditions {
+		if condition.Type == val.Type {
+			js.Status.Conditions[i] = condition
+			conditionExists = true
+		}
+	}
+	if !conditionExists {
+		js.Status.Conditions = append(js.Status.Conditions, condition)
+	}
 
 	if err := r.Status().Update(ctx, js); err != nil {
```

Comment:
We should update only if the condition changed

Reply:
Done.